dotnet/src/SemanticKernel.Abstractions/Data/VectorSearch/IVectorizedSearch.cs

```diff
@@ -1,24 +1,27 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
 using System.Threading;
 
 namespace Microsoft.SemanticKernel.Data;
 
 /// <summary>
-/// Interface for searching a vector store.
+/// Contains a method for doing a vector search using a vector.
 /// </summary>
 /// <typeparam name="TRecord">The record data model to use for retrieving data from the store.</typeparam>
-[Experimental("SKEXP0001")]
-public interface IVectorSearch<TRecord>
+public interface IVectorizedSearch<TRecord>
     where TRecord : class
 {
     /// <summary>
     /// Search the vector store for records that match the given embedding and filter.
     /// </summary>
-    /// <param name="vectorQuery">The vector to search with.</param>
+    /// <typeparam name="TVector">The type of the vector.</typeparam>
+    /// <param name="vector">The vector to search the store with.</param>
+    /// <param name="options">The options that control the behavior of the search.</param>
     /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
     /// <returns>The records found by the vector search, including their result scores.</returns>
-    IAsyncEnumerable<VectorSearchResult<TRecord>> SearchAsync(VectorSearchQuery vectorQuery, CancellationToken cancellationToken = default);
+    IAsyncEnumerable<VectorSearchResult<TRecord>> VectorizedSearchAsync<TVector>(
```

Comment:
If I want to page over the results how will I do that?

Reply:
You can use limit and offset together to do paging, e.g. for page two you can have this:
```csharp
await vectorSearch!.VectorizedSearchAsync(searchVector, new() { Limit = 3, Offset = 3, Filter = filter }).ToListAsync();
```